starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java

```diff
@@ -66,4 +67,47 @@ public void create() {
     Assert.assertTrue(inventoryPostInsertion.size() == existingInventory.size() + 1);
     Assert.assertTrue(createdInventory.getId().equals(returnedInventory.getId()));
   }
+  
+  /**
+   * Test Delete method with valid id.
+   */
+  @Test
+  public void delete() {
+    // Put an inventory in the databse so there's something to delete
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setProductType(PRODUCT_TYPE);
+    // Save the id upon insertion to check against when we delete it
+    Inventory savedInventory = mongoTemplate.save(inventory);
+    String targetId = savedInventory.getId();
+    // Count number of inventories in collection to make sure it's 1 less after a deletion
+    List<Inventory> existingInventories = this.mongoTemplate.findAll(Inventory.class);
+    // Delete an inventory, and check id of deleted inventory against the id we saved earlier
+    Optional<Inventory> deletedInventory = this.inventoryDAO.delete(targetId);
+    // Check size of collection post deletion
+    List<Inventory> inventoryPostDeletion = this.mongoTemplate.findAll(Inventory.class);
+    // Check that id of deleted inventory matches our target id
+    Assert.assertTrue(deletedInventory.map(Inventory::getId).orElse("default").equals(targetId));
```

Comment:
Instead of doing deletedInventory.map(Inventory::getID).orElse("default"), what is another way that we can access the id of the deletedInventory?

Reply:
I changed 
`Assert.assertTrue(deletedInventory.map(Inventory::getId).orElse("default").equals(targetId));`
to 
`Assert.assertTrue(deletedInventory.isPresent());`
`Assert.assertTrue(actualInventory.orElseThrow().getId().equals(inventory.getId()));`
But I have a question:
My new way, and all other ways I can think of to extract the id from the optional are all less concise and bulkier than my original way, which has a null check built-in with the orElse() method. From what I can tell other methods of extracting the id necessitate writing an explicit null check, making the code bulkier. Is there another more elegant way that I'm missing?

Reply:
You could do `Assert.assertTrue(actualInventory.get().getId().equals(inventory.getId()))` to access the object once you've verified that it isPresent.

Reply:
Hm the Javadocs and some articles I've found state that the `map()` + `orElse()` combination is preferred to the `isPresent()` and `orElseThrow()` combination, which is in turn preferred to `isPresent()` and `get()` combination.
The javadocs states that `orElseThrow` is the preferred alternative to `get`: https://github.com/pmd/pmd/issues/2719
and they are considering deprecating .get() to stop people from using it. Apparently Brian Goetz, who chose the name for the .get() method considers it "our biggest API mistake in Java8".
Here's an article that discusses using `orElse()` instead of `isPresent()` and `get()`: https://www.java67.com/2018/06/java-8-optional-example-ispresent-orElse-get.html
Scroll about halfway down to see the part that discusses `orElse()` vs `isPresent()` and `get()`
"Instead of using ifPresent() and get(), it's better to use OrElse() method which can return a default value if Optional doesn't contain a value."
Also, here's a really good youtube tutorial that discuss both of these comparisons starting at 11.41: https://youtu.be/vKVzRbsMnTQ?t=701

Reply:
Wow, thank you so much for your in-depth research and for providing it to me with an explanation! Go ahead and use what is recommended from those sources and I will share this with the rest of the mentors so they are aware of this as well.